src/index.d.ts

```diff
@@ -0,0 +1,158 @@
+declare module 'react-form' {
+  import {
+    ChangeEventHandler,
+    ComponentType,
+    Dispatch,
+    FocusEventHandler,
+    FormEventHandler,
+    HTMLProps,
+    Provider,
+    SetStateAction,
+  } from 'react'
+
+  type Debounce = <T>(fn: (...args: any[]) => T, wait: number) => Promise<T>
+  type ValidatorReturn = string | false | undefined
+  type OptionalPromise<T> = Promise<T> | T
+
+  interface FormMeta {
+    error: string | any
+    isSubmitting: boolean
+    isDirty: boolean
+    isSubmitted: boolean
+    submissionAttempts: number
+    isValid: boolean
+    fieldsAreValidating: boolean
+    fieldsAreValid: boolean
+    canSubmit: boolean
+    [k: string]: any
+  }
+
+  type SetFormMeta = Partial<FormMeta> | ((previousMeta: FormMeta) => FormMeta)
+
+  interface FormInstance<T> {
+    Form: ComponentType<Omit<HTMLProps<HTMLFormElement>, 'onSubmit'>>
+    values: T
+    meta: FormMeta
+    formContext: FormInstance<T>
+    reset: () => void
+    setMeta: (value: SetFormMeta) => void
+    handleSubmit: FormEventHandler<HTMLFormElement>
+    debounce: Debounce
+    setValues: Dispatch<SetStateAction<T>>
+    runValidation: () => void
+    getFieldValue: <V = any>(fieldPath: keyof T | string) => V
+    getFieldMeta: (
+      fieldPath: keyof T | string
+    ) => { error: string | null; [k: string]: any }
```

Comment:
```suggestion
    ) => FieldMeta
```

I suggest extracting this to have parity between `FormMeta` and `FieldMeta`. Also makes it easier to extend if more _react-form_ fields are added down the line.

**Edit**: There is a `FieldMeta`. Should this not be `FieldMeta`?

Reply:
You're totally right that type exists already. I believe that it's not guaranteed to have `isTouched` on the `FieldMeta` object, specifically when calling `getFieldMeta` though. The function returns the current meta, but that is initialized to an empty object https://github.com/tannerlinsley/react-form/blob/master/src/hooks/useForm.js#L256
https://github.com/tannerlinsley/react-form/blob/master/src/hooks/useForm.js#L17

Maybe `FieldMeta` should mark it as optional, just in case? Then we can reuse the interface.